src/MetadataWriter.vala

```diff
@@ -588,9 +581,10 @@ public class MetadataWriter : Object {
 
         if (dirty.contains (photo)) {
             bool removed = dirty.remove_first (photo);
-            assert (removed);
-
-            assert (!dirty.contains (photo));
+            if (!removed || dirty.contains (photo)) {
```

Comment:
Checking dirty seems redundant here since it is within an if-statement of the same check